rice-framework/krad-web-framework/src/main/java/org/kuali/rice/krad/web/filter/UserLoginFilter.java

```diff
@@ -147,11 +148,22 @@ private boolean isAuthorizedToLogin(String principalId) {
      * Creates a session id cookie if one does not exists.  Write the cookie out to the response with that session id.
      * Also, sets the cookie on the established user session.
      */
-    private void establishSessionCookie(HttpServletRequest request, HttpServletResponse response) {
+    protected void establishSessionCookie(HttpServletRequest request, HttpServletResponse response) {
         String kualiSessionId = this.getKualiSessionId(request.getCookies());
         if (kualiSessionId == null) {
             kualiSessionId = UUID.randomUUID().toString();
-            response.addCookie(new Cookie(KRADConstants.KUALI_SESSION_ID, kualiSessionId));
+            Cookie sessionCookie = new Cookie(KRADConstants.KUALI_SESSION_ID, kualiSessionId);
+
+            // Make cookie secure
+            sessionCookie.setSecure(true);
+
+            // Make cookie HttpOnly if supported by container
+            Method setHttpOnlyMethod = ReflectionUtils.findMethod(Cookie.class, "setHttpOnly", boolean.class);

```

Comment:
The Rice code is compiled against version 2.5 of the servlet API, so `setHttpOnly` isn't actually available on the Cookie class. However, the jar supplied by our Tomcat instances is providing a 3.x version of the servlet API where you _can_ `setHttpOnly` directly, so that's why we're able to do this at runtime.

I'm open to alternative suggestions to accomplishing this-- I just saw this method recommended in a couple places as a way of making cookies http only without dictating the specific version of the servlet API being provided.